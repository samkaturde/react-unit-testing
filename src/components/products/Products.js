import React from 'react'

const Products = ({ items = [] }) => {
        if (items.length <= 0) {
            return  <div>
                        No item found!
                    </div>
        } else {
            return <ol>
                {
                    items.map((item, ind) => {
                        return <li key={ind}>{item}</li>
                    })
                }
            </ol>
        }
}

export default Products;
