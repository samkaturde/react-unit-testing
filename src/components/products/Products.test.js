import React from "react";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import Products from "../products/Products";


afterEach(cleanup);

it('render when there is no item', () => {
    const tree = renderer.create(<Products />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('render when there is one item', () => {
    const tree = renderer.create(<Products items={['Apple']}/>).toJSON();
    expect(tree).toMatchSnapshot();
})


it('render when there are multiple items', () => {
    const tree = renderer.create(<Products items={['Apple', 'Mango', 'Banana']}/>).toJSON();
    expect(tree).toMatchSnapshot();
})