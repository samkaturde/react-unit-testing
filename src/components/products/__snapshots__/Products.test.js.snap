// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when there are multiple items 1`] = `
<ol>
  <li>
    Apple
  </li>
  <li>
    Mango
  </li>
  <li>
    Banana
  </li>
</ol>
`;

exports[`render when there is no item 1`] = `
<div>
  No item found!
</div>
`;

exports[`render when there is one item 1`] = `
<ol>
  <li>
    Apple
  </li>
</ol>
`;
